students/NatalieRodriguez/Lesson09/OO_Mailroom.py

```diff
@@ -0,0 +1,172 @@
+
+    def thank_you(self):
+        '''user inputs donor and amounts to dict.
+        '''
+        print('Type the full donor name (Enter "list" to show all donors or enter "dash" to return to the Donor Dashboard.')
+        new_donor = input(':')
+
+        if new_donor.lower() == 'list':
+            donor_names = []
+            for donor in self.donor_list:
+                donor_names.append(donor.name)
+            print(donor_names)
+
+        elif new_donor.lower() == 'dash':
+            return
+
+        else:
+            if new_donor in self.donor_list:
+                try:
+                    amount = float(input('Please enter the donation amount:'))
+                    donor_list[Donor].add_donation(amount)
+
+                except ValueError:
+                    print("Please enter a valid amount.")
+            else:
+                amount = float(input('Please enter the donation amount:'))
+                new_donor_object = Donor(new_donor, [amount])
+                self.donor_list.append(new_donor_object)
+                print('            Thank you {}, We greatly appreciate your generous donation of ${:.2f}.'.format(new_donor, amount))
+                print('''            Your contribution to the purchase of land for conservation
+            and habitat restoration will make a profound difference in allowing
+            the environment to recover from the detrimental impact of human
+            life. With noble individuals like yourself, we can make large strides
+            to allow the North American continent to exist without the stressors of 
+            agriculture and other human impacts. We appreciate and look forward to
+            your continued support.
+            
+            Best,
+            The Nature Conservancy''')
+
+    def donor_report(self):
+        '''Outputs a string table of donors and monetary amounts'''
+        print('Donor History')
+        report = []
+        report.append('|{:<20}|{:<20}|{:<20}|{:<20}|'.format('    Donor Name', '   Total Donation', '  No. of Donations', '  Average Donation'))
+        for donor in self.donor_list:
+           report.append('|{:<20}|{:>20}|{:>20}|{:>20}|'.format(donor.name, donor.total_donation(), donor.donation_amount(), donor.donation_avg()))
+        return '\n'.join(report)
+
+    def print_donors(self):
+        print(self.donor_report())
+
+    def write_file(self):
+        '''Creates a text file with a thank you message for each of the donors in the dictionary'''
+        for donor_data in self.donor_list:
+            filename = donor_data.name.replace(" ", "_") + ".txt"
+            total_donation = donor_data.total_donation()
+            letter = ('Thank you {} for you generous contributions in the amount of {:.2f}!'.format(donor_data.name, total_donation))
+            with open(filename, 'w') as file:
+                file.write(letter)
+            print(f"{donor_data.name}'s thank you has been saved at: " + filename)
```

Comment:
this can also live in Donor class; also, this function should be called `write_files` since you would be writing multiple files.